docs/data/charts/tooltip/ItemTooltip.tsx

```diff
@@ -0,0 +1,49 @@
+import * as React from 'react';
+import NoSsr from '@mui/material/NoSsr';
+import Popper from '@mui/material/Popper';
+import { useItemTooltip, useMouseTracker } from '@mui/x-charts/ChartsTooltip';
+import { CustomItemTooltipContent } from './CustomItemTooltipContent';
+import { generateVirtualElement } from './generateVirtualElement';
+
+export function ItemTooltip() {
+  const tooltipData = useItemTooltip();
+  const mousePosition = useMouseTracker(); // Track the mouse position on chart.
+
+  if (!tooltipData || !mousePosition) {
+    // No data to display
+    return null;
+  }
+
+  return (
+    <NoSsr>
+      <Popper
+        sx={{
+          pointerEvents: 'none',
+          zIndex: (theme) => theme.zIndex.modal,
+        }}
+        open
+        placement={
+          mousePosition?.pointerType === 'mouse'
+            ? ('top-end' as const)
+            : ('top' as const)
+        }
+        anchorEl={generateVirtualElement(mousePosition)}
+        modifiers={[
+          {
+            name: 'offset',
+            options: {
+              offset: [
+                0,
+                mousePosition?.pointerType === 'touch'
+                  ? 40 - mousePosition.height
+                  : 0,
+              ],
+            },
+          },
+        ]}
```

Comment:
```suggestion
        anchorEl={generateVirtualElement(mousePosition)}
```

What do you think about leaving these out to keep this simple?

Reply:
I kept the behavior, but refactore it and added comments to keep the information and make it easier to hunderstand